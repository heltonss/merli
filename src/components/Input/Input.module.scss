@import '../../assets/scss/colors';

.container {
  width: 100%;
  height: 36px;
  padding-left: 16px;

  &::placeholder {
    color: $gray;
  }

  &::-ms-input-placeholder {
    color: $gray;
  }

  &::-ms-input-placeholder {
    color: $gray;
  }
}
