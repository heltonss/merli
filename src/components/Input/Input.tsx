import classnames from 'classnames';
import React from 'react';
import Style from './Input.module.scss';

interface IOwnProps {
  placeholder: string;
  type: string;
  parentClassStyle?: string;
  getValue(value: string): void;
}

const Input = ({ placeholder, type, getValue, parentClassStyle }: IOwnProps) => {
  const getValueInput = (event: any) => {
    getValue(event.target.value);
  };
  return (
    <input
      className={classnames([Style.container, parentClassStyle])}
      type={type}
      placeholder={placeholder}
      onChange={getValueInput}
    />
  );
};

export default Input;
