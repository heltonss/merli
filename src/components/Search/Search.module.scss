@import '../../assets/scss/colors';
@import '../../assets/scss/screen';

.container {
  width: 100%;
  display: flex;
  align-items: center;
}

.input {
  border-top-left-radius: 4px;
  border-bottom-left-radius: 4px;
}
