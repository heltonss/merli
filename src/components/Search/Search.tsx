import IconSearch from 'components/Icons/IconSearch/IconSearch';
import Input from 'components/Input/Input';
import React, { useState } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';
import { Creators as searchProducts } from 'store/ducks/search';
import Style from './Search.module.scss';

interface IOwnProps {
  searchProducts(product: string): void;
}

const Search: React.FC = (props: any) => {
  const [valueSearch, setValueSearch] = useState('');

  const getValueInput = (value: string) => {
    setValueSearch(value);
  };

  const search = () => {
    props.searchProducts(valueSearch);
  };

  return (
    <div className={Style.container}>
      <Input
        parentClassStyle={Style.input}
        type={'text'}
        placeholder={'Buscar produtos, marcas e muito mais...'}
        getValue={getValueInput}
        onEnter={search}
      />
      <IconSearch func={search} isRounded={true} />
    </div>
  );
};

const mapDispatchToProps = (dispatch: Dispatch) => bindActionCreators(searchProducts, dispatch);

export default connect(
  undefined,
  mapDispatchToProps,
)(Search);
