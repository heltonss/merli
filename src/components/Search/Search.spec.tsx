import { shallow } from 'enzyme';
import React from 'react';
import Input from '../Input/Input';
import Search from './Search';

describe('Search', () => {
  it('Should render withou crash', () => {
    const wrapper = shallow(<Search />);

    expect(wrapper).toMatchSnapshot();
  });
  it('Should have one input', () => {
    const wrapper = shallow(<Search />);

    expect(wrapper.find(Input)).toHaveLength(1);
  });
});
