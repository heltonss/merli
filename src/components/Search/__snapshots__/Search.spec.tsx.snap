// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Should render withou crash 1`] = `ShallowWrapper {}`;
